analysis_driver/report_generation/report_crawlers.py

```diff
@@ -14,119 +14,134 @@
-                        barcode_info[ELEMENT_RUN_NAME]=self.run_id
-                        barcode_info[ELEMENT_PROJECT] = 'default'
-                        barcode_info[ELEMENT_SAMPLE_INTERNAL_ID]='Undetermined'
-                        barcode_info[ELEMENT_LIBRARY_INTERNAL_ID]='Undetermined'
-                        barcode_info[ELEMENT_LANE]=lane
-                        self.barcodes_info[barcode_info[ELEMENT_RUN_ELEMENT_ID]]=(barcode_info)
+                        barcode_info = {
+                            ELEMENT_BARCODE: 'unknown',
+                            ELEMENT_RUN_ELEMENT_ID: '%s_%s_%s' % (self.run_id, lane, 'unknown'),
+                            ELEMENT_RUN_NAME: self.run_id,
+                            ELEMENT_PROJECT: 'default',
+                            ELEMENT_SAMPLE_INTERNAL_ID: 'Undetermined',
+                            ELEMENT_LIBRARY_INTERNAL_ID: 'Undetermined',
+                            ELEMENT_LANE: lane
+                        }
+                        self.barcodes_info[barcode_info[ELEMENT_RUN_ELEMENT_ID]] = barcode_info
         self.run[ELEMENT_NUMBER_LANE] = len(self.lanes)
-        self.unexpected_barcode_info={}
 
     def _populate_barcode_info_from_conversion_file(self, conversion_xml_file):
-        all_barcodes_per_lanes, top_unknown_barcodes_per_lanes = parse_conversion_stats(conversion_xml_file)
-        nb_read_per_lane=Counter()
-        for project, library, lane, barcode, clust_count, clust_count_pf, nb_bases,\
-            nb_bases_r1q30, nb_bases_r2q30, in all_barcodes_per_lanes:
-            barcode_info = self.barcodes_info.get('%s_%s_%s'%(self.run_id, lane, barcode))
-            barcode_info[ELEMENT_NB_READS_SEQUENCED]=int(clust_count)
-            barcode_info[ELEMENT_NB_READS_PASS_FILTER]=int(clust_count_pf)
-            nb_read_per_lane[lane]+=int(clust_count_pf)
-            #For the paired end reads
-            barcode_info[ELEMENT_NB_BASE_R1]=int(nb_bases)
-            barcode_info[ELEMENT_NB_BASE_R2]=int(nb_bases)
-            barcode_info[ELEMENT_NB_Q30_R1]=int(nb_bases_r1q30)
-            barcode_info[ELEMENT_NB_Q30_R2]=int(nb_bases_r2q30)
+        all_barcodes_per_lanes, top_unknown_barcodes_per_lanes = demultiplexing_parsers.parse_conversion_stats(conversion_xml_file)
+        nb_read_per_lane = Counter()
+        for project, library, lane, barcode, clust_count, clust_count_pf, nb_bases, nb_bases_r1_q30, nb_bases_r2_q30 in all_barcodes_per_lanes:
+            nb_read_per_lane[lane] += int(clust_count_pf)
+            barcode_info = self.barcodes_info.get('%s_%s_%s' % (self.run_id, lane, barcode))
+            barcode_info[ELEMENT_NB_READS_SEQUENCED] = int(clust_count)
+            barcode_info[ELEMENT_NB_READS_PASS_FILTER] = int(clust_count_pf)
+            # For the paired end reads
+            barcode_info[ELEMENT_NB_BASE_R1] = int(nb_bases)  # TODO: is this correct?
```

Comment:
Yes it is, it assumes that read1 and read2 have the same read length which is always the case.